components/icon/src/build.js

```diff
@@ -1,94 +1,97 @@
   string: ['prefix', 'path'],
 });
+const { readdir, readFile, writeFile, mkdir } = require('node:fs/promises');
 const path = require('path');
 const { component, test, generateIconsName, style, iconContainerComponent } = require('./component_templates.js');
 
 const { kebabCase } = lodash;
 const componentPrefix = `${argv.prefix}-`;
 const componentsPath = argv.path;
 const iconRapperComponent = `${componentPrefix}container`;
-
-if (fs.existsSync(componentsPath)) throw new Error(`A component with this name already exists: ${componentsPath}`);
-
-if (!componentsPath) throw new Error('You must include a components path name.');
-
-fs.mkdirSync(componentsPath, { recursive: true });
-
-function writeFileErrorHandler(err) {
-  if (err) throw err;
-}
-
-const directoryPath = path.join(__dirname, '../tmp/optimized-svgs');
-
-fs.readdir(directoryPath, function (err, files) {
-  //handling error
-  if (err) {
-    console.error('Unable to scan directory: ' + err);
-    return;
+const cwd = path.resolve(process.cwd(), 'tmp/optimized-svgs');
+
+const readSVGDir = async () => {
+  let files;
+  try {
+    files = await readdir(cwd);
+    return files;
+  } catch (error) {
+    console.error(error);
   }
+  return files;
+};
 
-  const iconData = files.map((file) => ({
-    id: `${componentPrefix}${kebabCase(file.replace('.svg', ''))}`,
-    src: file,
-  }));
-
-  files.forEach(function (file) {
-    const fileName = `${componentPrefix}${kebabCase(file.replace('.svg', ''))}`;
-    const formattedName = kebabCase(fileName);
-    const dir = `${componentsPath}/${formattedName}`;
-
-    fs.readFile(path.join(directoryPath, file), 'utf8', function (err, svg) {
-      if (err) {
-        console.error(err);
-        return;
-      }
-
-      // throw an error if the file already exists
-      if (fs.existsSync(dir)) throw new Error('A component with that name already exists.');
-
-      // create the folder
-      fs.mkdirSync(dir);
-      fs.writeFile(
-        `${dir}/${formattedName}.stencil.tsx`,
-        component(fileName, svg, iconRapperComponent),
-        writeFileErrorHandler,
+const getIconData = async () => {
+  let svgData;
+
+  const files = await readSVGDir();
+  try {
+    if (files && files.length > 0) {
+      svgData = await Promise.all(
+        files.map((file) => ({
+          id: `${componentPrefix}${kebabCase(path.parse(file).name)}`,
+          src: file,
+        })),
       );
-      fs.writeFile(`${dir}/${formattedName}.spec.tsx`, test(fileName), writeFileErrorHandler);
-    });
-  });
-
-  const dirExistsSync = (dir) => {
-    const stats = !!fs.statSync(dir, { throwIfNoEntry: false });
-    return stats ? stats.isDirectory() : false;
-  };
-
-  const outputDir = path.resolve(__dirname, '../dist/');
+    }
+  } catch (error) {
+    console.error(error);
+  }
+  return svgData;
+};
 
-  if (!dirExistsSync(outputDir)) {
-    fs.mkdirSync(outputDir);
+const WriteComponentFile = async () => {
```

Comment:
Starting `WriteComponentFile` with a capital is quite unusual for a function, mostly we use that for classes.

Reply:
You are right, it's a typo